app/components/wds-global-navigation/search.js

```diff
@@ -29,7 +59,247 @@ export default Component.extend({
+
+		fetch(uri)
+			.then((response) => {
+				if (response.ok) {
+					if (this.get('isDestroyed')) {
+						return;
+					}
+
+					return response.json().then((data) => {
+						const suggestions = data.suggestions.map((suggestion) => {
+							return EmberObject.create({
+								title: suggestion
+							});
+						});
+
+						/**
+						 * If the user makes one request, request A, and then keeps typing to make
+						 * request B, but request A takes a long time while request B returns quickly,
+						 * then we don't want request A to dump its info into the window after B has
+						 * already inserted the relevant information.
+						 * Also, we don't want to show the suggestion results after a real search
+						 * will be finished, what will happen if search request is still in progress.
+						 */
+						if (!this.get('searchRequestInProgress') && query === this.get('query')) {
+							this.setSearchSuggestionItems(suggestions);
+						}
+
+						this.cacheResult(query, suggestions);
+					});
+				} else if (response.status === 404) {
+					// When we get a 404, it means there were no results
+					if (query === this.get('query')) {
+						this.setSearchSuggestionItems();
+					}
+
+					this.cacheResult(query);
+				} else {
+					this.get('logger').error('Search suggestions error', response);
+				}
+			})
+			.catch((reason) => this.get('logger').error('Search suggestions error', reason))
+			.finally(() => {
+				// We have a response, so we're no longer loading the results
+				if (query === this.get('query') && !this.get('isDestroyed')) {
+					this.set('isLoadingResultsSuggestions', false);
+				}
+
+				this.endedRequest(query);
+			});
+	},
+
+	/**
+	 * Wrapper for search suggestions performing, that also checks the cache
+	 */
+	getSuggestions(query) {
+		if (this.get('isDestroyed')) {
+			return;
+		}
+
+		this.setProperties({
+			suggestions: [],
+			searchRequestInProgress: false
+		});
+
+		// If the query string is empty or shorter than the minimal length, return to leave the view blank
+		if (!query || query.length < this.get('queryMinimalLength')) {
+			this.set('isLoadingResultsSuggestions', false);
+		} else if (this.hasCachedResult(query)) {
+			this.setSearchSuggestionItems(this.getCachedResult(query));
+		} else {
+			this.set('isLoadingResultsSuggestions', true);
+			run.debounce(this, () => this.requestSuggestionsFromAPI(query), this.get('debounceDuration'));
+		}
+	},
+
+	normalizeToUnderscore(title = '') {
+		return title
+			.replace(/\s/g, '_')
+			.replace(/_+/g, '_');
+	},
+
+	escapeRegex(text) {
+		return text.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&');
+	},
+
+	/**
+	 * @param {SearchSuggestionItem[]} [suggestions = []]
+	 * @returns {void}
+	 */
+	setSearchSuggestionItems(suggestions = []) {
+		if (this.get('isDestroyed')) {
+			return;
+		}
+
+		const query = this.get('query'),
+			highlightRegexp = new RegExp(`(${this.escapeRegex(query)})`, 'ig'),
+			highlighted = wrapMeHelper.compute(['$1'], {
```

Comment:
as we use wrapMe helper only in JS it doesn't need to be helper just a simple function. unless we want to share these helper to applications and use there in hbs.

Reply:
AFAIR @hakubo told me that wrapMe will be extracted to ember-fandom

Reply:
Assuming it's resolved 👍